wombat_ic_generation/regrid_forcing.py

```diff
@@ -0,0 +1,193 @@
+# Interpolate a provided forcing file to a grid specified by a provided MOM supergrid file for
+# use in the MOM data_table.
+#
+# To run:
+#   python regrid_forcing.py --forcing-filename=<path-to-forcing-file>
+#       --hgrid-filename=<path-to-supergrid-file> --output-file=<path-to-output-file>
+# these extra arguments are available if required:
+#     --lon-name=<lon-name> --lat-name=<lat-name>
+#
+# For more information, run `python regrid_forcing.py -h`
+#
+# The run command and full github url of the current version of this script is added to the
+# metadata of the generated file. This is to uniquely identify the script and inputs used to
+# generate the file. To produce files for sharing, ensure you are using a version of this script
+# which is committed and pushed to github. For files intended for released configurations, use the
+# latest version checked in to the main branch of the github repository.
+#
+# Contact:
+#   Dougie Squire <dougal.squire@anu.edu.au>
+#
+# Dependencies:
+#   argparse, xarray, cf_xarray, xesmf
+# =========================================================================================
+
+import os
+import sys
+from pathlib import Path
+
+import xarray as xr
+import xesmf as xe
+
+path_root = Path(__file__).parents[1]
+sys.path.append(str(path_root))
+
+from scripts_common import get_provenance_metadata, md5sum
+
+
+def _guess_longitude_name(ds):
+    coords = ds.cf.coordinates
+    if "longitude" in coords:
+        if len(coords["longitude"]) == 1:
+            return coords["longitude"][0]
+        else:
+            raise KeyError("Multiple longitude variables exist. Please pass --lon-name")
+    elif "lon" in ds:
+        return "lon"
+    else:
+        raise KeyError(
+            "Cannot automatically determine the name of the longitude variable. Please pass --lon-name"
+        )
+
+
+def _guess_latitude_name(ds):
+    coords = ds.cf.coordinates
+    if "latitude" in coords:
+        if len(coords["latitude"]) == 1:
+            return coords["latitude"][0]
+        else:
+            raise KeyError("Multiple latitude variables exist. Please pass --lat-name")
+    elif "lat" in ds:
+        return "lat"
+    else:
+        raise KeyError(
+            "Cannot automatically determine the name of the latitude variable. Please pass --lat-name"
+        )
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description=(
+            "Interpolate a provided forcing file to a grid specified by a provided MOM supergrid"
+            "file for use in the MOM data_table."
+        )
+    )
+
+    parser.add_argument(
+        "--forcing-filename",
+        required=True,
+        help="The path to the forcing file to interpolate.",
+    )
+
+    parser.add_argument(
+        "--hgrid-filename",
+        required=True,
+        help="The path to the MOM supergrid file to use as a grid for interpolation.",
+    )
+
+    parser.add_argument(
+        "--output-file",
```

Comment:
```suggestion
        "--output-filename",
```

filename would be consistent with the other arguments